components/Common/AvatarGroup/index.tsx

```diff
@@ -0,0 +1,62 @@
+'use client';
+import * as Avatar from '@radix-ui/react-avatar';
+import { useState } from 'react';
+import type { FC } from 'react';
+import styles from './index.module.css';
+
+type AvatarGroupProps = {
+  avatars: {
+    src: string;
+    name: string;
+  }[];
+  limit?: number;
+};
+
+const userNameToFallBackText = (name: string) => {
+  // sanitize name before using it
+  name = name.toLowerCase().replace(' ', '-');
+  const words = name.split('-');
+  if (words.length > 1) {
+    return words[0][0] + words[1][0];
+  }
+  return name.slice(0, 2);
+};
+
+const AvatarGroup: FC<AvatarGroupProps> = ({ avatars, limit = 10 }) => {
+  const [showMore, setShowMore] = useState(false);
+
+  const toggleShowMore = () => {
+    setShowMore(!showMore);
+  };
+
+  console.log('avatars', avatars);
+
+  const renderAvatars = showMore ? avatars : avatars.slice(0, limit);
+
+  return (
+    <div className={styles.avatarGroup}>
+      {renderAvatars.map((avatar, index) => (
+        <Avatar.Root key={index} className={styles.avatarRoot}>
```

Comment:
You should make an individual component for Avatar, as the issue states. Even tho you're using RadixUI.

Please follow the issue description...